LibWeb: Crash with a transform animation that contains var()

Consider the following snippet: 

```html
<style>
    div {
        --some-var: -100px;
        animation: anim 1s infinite;
    }

    @keyframes anim {
        to {
            transform: translateX(calc(var(--some-var) * -1));
        }
    }
</style>
<div>hello world</div>
```

Before #23622, this would just do nothing. The value of `transform` would be an `UnresolvedStyleValue` due to the variable, and we wouldn't do anything about it. However, after that PR, we now resolve it to the following:

```html
<style>
    div {
        animation: anim 1s infinite;
    }

    @keyframes anim {
        to {
            transform: translateX(calc(-100px * -1));
        }
    }
</style>
<div>hello world</div>
```

Now it is no longer unresolved, but crashes for some reason! The second snippet crashed both before and after that PR with the same trace, however it is now visible on apple.com. Note that this does not crash if the `transform` is applied to the element directly instead of through an animation.

<details><summary>Crash log</summary>
<p>

```
-- Configuring done (1.0s)
-- Generating done (0.3s)
-- Build files have been written to: /home/matthew/code/serenity/Build/lagom
ninja: Entering directory `/home/matthew/code/serenity/Build/lagom'
[0/2] Re-checking globbed directories...
[0/1] cd /home/matthew/code/serenity/Build/lagom/Ladybird && /usr/bin/cmake -E env SERENITY_SOURCE_DI.../serenity/Build/lagom/bin/Ladybird /home/matthew/code/serenity/Base/res/html/misc/web-animations.html
2605.339 RequestServer(55604): Loaded 145 of 145 (100%) provided CA Certificates
VERIFICATION FAILED: m_pointer at /home/matthew/code/serenity/Meta/Lagom/../../AK/Optional.h:436
/home/matthew/code/serenity/Build/lagom/lib/liblagom-ak.so.0(ak_verification_failed+0xef) [0x7d3f1193504f]
/home/matthew/code/serenity/Build/lagom/lib/liblagom-web.so.0 Web::CSS::CalculatedStyleValue::CalculationResult::multiply_by(Web::CSS::CalculatedStyleValue::CalculationResult const&, AK::Optional<Web::CSS::Length::ResolutionContext const&>) 0x4c7) [0x7d3f120041b7]
/home/matthew/code/serenity/Build/lagom/lib/liblagom-web.so.0 Web::CSS::ProductCalculationNode::resolve(AK::Optional<Web::CSS::Length::ResolutionContext const&>, AK::Variant<AK::Empty, Web::CSS::Angle, Web::CSS::Flex, Web::CSS::Frequency, Web::CSS::Length, Web::CSS::Time> const&) const 0xb3) [0x7d3f12003b43]
/home/matthew/code/serenity/Build/lagom/lib/liblagom-web.so.0 Web::CSS::CalculatedStyleValue::resolve_angle() const 0x4e) [0x7d3f1200f84e]
/home/matthew/code/serenity/Build/lagom/lib/liblagom-web.so.0 Web::CSS::AngleOrCalculated::resolve_calculated(AK::NonnullRefPtr<Web::CSS::CalculatedStyleValue> const&, Web::Layout::Node const&) const 0x24) [0x7d3f11f55524]
/home/matthew/code/serenity/Build/lagom/lib/liblagom-web.so.0(+0x628407) [0x7d3f12028407]
/home/matthew/code/serenity/Build/lagom/lib/liblagom-web.so.0 Web::CSS::Transformation::to_matrix(AK::Optional<Web::Painting::PaintableBox const&>) const 0x868) [0x7d3f120278b8]
/home/matthew/code/serenity/Build/lagom/lib/liblagom-web.so.0(+0x5eaa23) [0x7d3f11feaa23]
/home/matthew/code/serenity/Build/lagom/lib/liblagom-web.so.0(+0x5e9a20) [0x7d3f11fe9a20]
/home/matthew/code/serenity/Build/lagom/lib/liblagom-web.so.0 Web::CSS::StyleComputer::collect_animation_into(Web::DOM::Element&, AK::Optional<Web::CSS::Selector::PseudoElement::Type>, JS::NonnullGCPtr<Web::Animations::KeyframeEffect>, Web::CSS::StyleProperties&, Web::CSS::StyleComputer::AnimationRefresh) const 0x75d) [0x7d3f11fde77d]
/home/matthew/code/serenity/Build/lagom/lib/liblagom-web.so.0 Web::Animations::KeyframeEffect::update
_style_properties() 0xe2) [0x7d3f11f0bc02]
/home/matthew/code/serenity/Build/lagom/lib/liblagom-web.so.0 Web::DOM::Document::update_style() 0x1d1) [0x7d3f1203d471]
/home/matthew/code/serenity/Build/lagom/lib/liblagom-web.so.0 Web::DOM::Document::update_layout() 0x8c) [0x7d3f1203cd4c]
/home/matthew/code/serenity/Build/lagom/lib/liblagom-web.so.0 Web::HTML::EventLoop::process() 0x29b) [0x7d3f12143cfb]
/home/matthew/code/serenity/Build/lagom/lib/liblagom-web.so.0(+0x990a40) [0x7d3f12390a40]
/home/matthew/code/serenity/Build/lagom/lib/liblagom-core.so.0(+0x3a0ff) [0x7d3f119c50ff]
/home/matthew/code/serenity/Build/lagom/lib/liblagom-core.so.0 Core::EventReceiver::dispatch_event(Core::Event&, Core::EventReceiver*) 0x75) [0x7d3f119c45d5]
/home/matthew/code/serenity/Build/lagom/libexec/WebContent(+0xdf621) [0x5fb1ef3a1621]
/usr/lib/libQt6Core.so.6(+0x190ca9) [0x7d3f13790ca9]
/usr/lib/libQt6Core.so.6 QTimer::timerEvent(QTimerEvent*) 0xa4) [0x7d3f13798d64]
/usr/lib/libQt6Core.so.6 QObject::event(QEvent*) 0x1b6) [0x7d3f1377c236]
/usr/lib/libQt6Core.so.6 QCoreApplication::notifyInternal2(QObject*, QEvent*) 0x15c) [0x7d3f1373979c]
/usr/lib/libQt6Core.so.6 QTimerInfoList::activateTimers() 0x516) [0x7d3f1389efa6]
/usr/lib/libQt6Core.so.6(+0x375724) [0x7d3f13975724]
/usr/lib/libglib-2.0.so.0(+0x59f69) [0x7d3f0f720f69]
/usr/lib/libglib-2.0.so.0(+0xb83a7) [0x7d3f0f77f3a7]
/usr/lib/libglib-2.0.so.0(g_main_context_iteration+0x32) [0x7d3f0f71f162]
/usr/lib/libQt6Core.so.6 QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) 0x84) [0x7d3f139739c4]
/usr/lib/libQt6Core.so.6 QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) 0x19e) [0x7d3f13743d6e]
/home/matthew/code/serenity/Build/lagom/lib/liblagom-core.so.0 Core::EventLoop::exec() 0x31) [0x7d3f119bd8d1]
/home/matthew/code/serenity/Build/lagom/libexec/WebContent(+0xe8746) [0x5fb1ef3aa746]
/home/matthew/code/serenity/Build/lagom/libexec/WebContent(main+0xf2) [0x5fb1ef3ac252]
/usr/lib/libc.so.6(+0x25cd0) [0x7d3f10843cd0]
/usr/lib/libc.so.6(__libc_start_main+0x8a) [0x7d3f10843d8a]
/home/matthew/code/serenity/Build/lagom/libexec/WebContent(+0x3a105) [0x5fb1ef2fc105]
2605.582 Ladybird(55598): WebContent process crashed!
2605.626 RequestServer(55620): Loaded 145 of 145 (100%) provided CA Certificates
^Cninja: build stopped: interrupted by user.
```
</p>
</details>

--- a/Tests/LibWeb/Text/expected/css/animating-transform-with-calc-crash.txt
+++ b/Tests/LibWeb/Text/expected/css/animating-transform-with-calc-crash.txt
@@ -1,0 +1,1 @@
+   PASS! (Didn't crash)

--- a/Tests/LibWeb/Text/input/css/animating-transform-with-calc-crash.html
+++ b/Tests/LibWeb/Text/input/css/animating-transform-with-calc-crash.html
@@ -1,0 +1,21 @@
+<!-- https://github.com/SerenityOS/serenity/issues/23633 -->
+<style>
+    div {
+        --some-var: -100px;
+        animation: anim 1s infinite;
+    }
+
+    @keyframes anim {
+        to {
+            transform: translateX(calc(var(--some-var) * -1));
+        }
+    }
+</style>
+<div></div>
+<script src="../include.js"></script>
+<script>
+    promiseTest(async () => {
+        await animationFrame();
+        println("PASS! (Didn't crash)");
+    });
+</script>

--- a/Userland/Libraries/LibWeb/CSS/StyleComputer.cpp
+++ b/Userland/Libraries/LibWeb/CSS/StyleComputer.cpp
@@ -870,7 +870,7 @@ static RefPtr<StyleValue const> interpolate_transform(DOM::Element& element, Sty
                 values.append(AngleOrCalculated { value->as_angle().angle() });
                 break;
             case StyleValue::Type::Calculated:
-                values.append(AngleOrCalculated { value->as_calculated() });
+                values.append(LengthPercentage { value->as_calculated() });
                 break;
             case StyleValue::Type::Length:
                 values.append(LengthPercentage { value->as_length().length() });

